sacro/__init__.py

```diff
@@ -1,13 +1,26 @@
+import logging
+import os
+
 import uvicorn
-from fastapi import FastAPI
+from fastapi import Cookie, FastAPI, HTTPException
 from fastapi.responses import HTMLResponse
 
 
+logger = logging.getLogger(__name__)
 app = FastAPI()
+ENV_TOKEN = os.environ.get("SACRO_APP_TOKEN")
+PORT = int(os.environ.get("PORT", "8000"))
+
+if ENV_TOKEN:
+    logger.info("running with SACRO_APP_TOKEN")
 
 
 @app.get("/")
-async def root():
+async def root(sacro_app_token: str | None = Cookie(default=None)):
```

Comment:
Is this FastAPI being clever and working out the name of the cookie from the name of the kwarg it's being assigned to?

Reply:
Yup. Not a huge fan, tbh.  

I just used fastapi as that's what release hatch uses, we may just go django something else. I just wanted to use something with some c deps (pydantic) so I could test that we can build/ship C libs.